pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/invertedindex/RealtimeInvertedIndexReader.java

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.core.realtime.impl.invertedindex;
+
+import com.linkedin.pinot.common.utils.Pairs;
+import com.linkedin.pinot.core.segment.index.readers.InvertedIndexReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import org.roaringbitmap.buffer.ImmutableRoaringBitmap;
+import org.roaringbitmap.buffer.MutableRoaringBitmap;
+
+
+public class RealtimeInvertedIndexReader implements InvertedIndexReader {
+  private final List<ThreadSafeMutableRoaringBitmap> _bitmaps = new ArrayList<>();
+  private final ReentrantReadWriteLock.ReadLock _readLock;
+  private final ReentrantReadWriteLock.WriteLock _writeLock;
+
+  public RealtimeInvertedIndexReader() {
+    ReentrantReadWriteLock readWriteLock = new ReentrantReadWriteLock();
+    _readLock = readWriteLock.readLock();
+    _writeLock = readWriteLock.writeLock();
+  }
+
+  /**
+   * Add the document id to the bitmap for the given dictionary id.
+   */
+  public void add(int dictId, int docId) {
+    if (_bitmaps.size() == dictId) {
```

Comment:
Add a comment that this check is for new entry to the list.

Reply:
Next line is the documentation :-P